examples/subcurrency/src/main.sw

```diff
@@ -90,15 +90,6 @@ impl Token for Contract {
         // Note: The return type of `msg_sender()` can be inferred by the
         // compiler. It is shown here for explicitness.
         let sender: Result<Identity, AuthError> = msg_sender();
```

Comment:
We still need to extract the `Address` from `msg_sender()` because the `balances` expects an `Address` as its key, right? We probably still need the `match` statement but without the `assert`. Something like:
```
        let sender: Address = match sender.unwrap() {
            Identity::Address(addr) => {
                addr
            },
            _ => {
                revert(0);
            },
        };
```

Reply:
Thanks! Just added the requested changes.